### DISCUSSION TITLE
Ideas/proposals around having mining pools

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#4005] [General Q&A]
It will be nice to have a way for smaller miners to joint their power/storage space/hardware together to have a greater chance to mine a block in the network.

This is a discussion to gather any genius or crazy ideas/thoughts from our community members how can we potentially achieve this goal.

**Comment by jennijuju** [comment]
from Chris

I think that a lot of small miners have definitely be thinking of something like this. I think the group rules need a little more flushing out for sure.  I think the idea should be to help miners grow to a point where they no longer need the group, so hopefully you are never forcing anyone out, but they are leaving because the group helped them grow to a point where they could compete on Mainnet on their own. IMO

**Comment by TippyFlitsUK** [comment]
These guys seem to be planning something [HDPool](https://hdpool.com).

They also link directly to a site selling [Filecoin Cloud Mining](https://ttbit.com/#/)...

**Reply by B1aZer** [reply]
Really like the idea of pool.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
is the idea to make it 1x4% instead of 2x2% per election? if this mathematically really increases the chance of winning a block - i have no time to calc that right now - if yes, we have a problem implementing this


miners can already use the same owner address to pool their rewards!

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
let me elaborate on this answer a bit since it might now be clear what i mean from the short version i wrote between gettin' up and bringing the kids to school.

> It will be nice to have a way for smaller miners to joint their power/storage space/hardware together to have a greater chance to mine a block in the network.

**Scenario 1: joining Power**

if we have 2 miners with each 1% of adjusted power and we combine them to 2% and we end up with a greater chance to mine a block, this cannot be implemented. Why?

- best case: we end up with a single pool mining 100% of the blocks.
- worst case: we have bigger pools preying on smaller pools, solo miners

if combining power doesn't end up in more blocks - whats the sense of doing it? sure, leveling rewards --> see "joining/pooling rewards" below

Ok! Let's limit the size of miner (power) who can participate in a pool (as fraud an argument as it has ever been, the size is not a good measure to do anything in regards to regulate miners). The case that makes sense in regard to limiting by miner size  is circumventing the 10TB rule - but is that wanted? (And yes the 10TB rule is as bad as every other "limit stuff by miner size" rule!!!)

**Scenario 2: pooling/joining hardware**

Totally possible, sharing/selling sealing pipelines, selling shares in mining operations (the limit here is your imagination and your imagination only, maybe a few tech issues on the way but besides that: go and get creative!) - it's all been done. The overhead is more or less expensive depending on how this is organized. 

Important is: it's possible, no one stops you now - this is nothing that needs to be implemented on chain.

**Scenario 3: joining/pooling rewards**

Again, possible via shared owner addresses. It is not being done, as far as i know (across different, separated business entities - miner layouts like 6 block's will surely do it - search on chain, it will be there.), because the reward distribution logic that comes with it is out of control complex. It would need trust, and a killer logic to manage stuff like slashing, missing wdPosts, deals etc. 

Can be done totally off chain, getting this into a "pool actor" will blow up the chain in size (in addition we first would need an agreement on how this should work at all, ending up with possibly a programmable actor...)



We are discussing pooling hardware, pooling rewards since day -XYZ, long before mainnet in smaller, bigger groups along the way. It is possible, no one stops miners from doing it (hardware, rewards pool) except for explicitly joining power - but if that would bring an advantage it is nothing that can be implemented.

**Comment by mxz9102** [comment]
The miner pool sound a great idea. I noticed the lucky ratio is going down now on my miner as the percentage of its power goes down.
